### ISSUE TITLE
ensure that IsCancun is used as much as possible

### ISSUE THREAD

**Issue by gballet** [gballet/go-verkle#166]
In many instances, `evm.txContext.Accesses != nil` can be replaced by a call to `IsCancun()`, we should strive for using this, in order to make it easier for reviewers to figure out that this code will not activate outside of a verkle tree-enabled context, even if `Accesses` are somehow activated.

**Comment by gballet** [comment]
@jwasinger did most of the work, there are a couple issues left in #44/#60:

```
core/vm/evm.go:	if evm.Accesses != nil && value.Sign() != 0 {
core/vm/evm.go:	if err == nil && evm.Accesses != nil {
```

**Comment by gballet** [comment]
Solved with 4087d8b
